api/jobs/gears.py

```diff
@@ -54,38 +53,24 @@ def get_gear_by_name(name):
 def get_invocation_schema(gear):
     return gear_tools.derive_invocation_schema(gear['gear'])
 
-def suggest_container(gear, cont_name, cid):
+def add_suggest_info_to_files(gear, files):
     """
-    Given a container reference, suggest files that would work well for each input on a gear.
+    Given a list of files, add information to each file that details those that would work well for each input on a gear.
     """
 
-    root = ContainerStorage.factory(cont_name).get_container(cid, projection={'permissions':0}, get_children=True)
-    root['analyses'] = ContainerStorage.factory('analyses').get_analyses(cont_name, cid, False)
-
     invocation_schema = get_invocation_schema(gear)
 
     schemas = {}
     for x in gear['gear']['inputs']:
         schema = gear_tools.isolate_file_invocation(invocation_schema, x)
         schemas[x] = Draft4Validator(schema)
 
-    # It would be nice to have use a visitor here instead of manual key loops.
-    for acq in root.get('acquisitions', []):
-        for f in acq.get('files', []):
-            f['suggested'] = {}
-            for x in schemas:
-                f['suggested'][x] = schemas[x].is_valid(f)
-
-    for analysis in root.get('analyses',[]):
-        files = analysis.get('files', [])
-        files[:] = [x for x in files if x.get('output')]
-        for f in files:
-            f['suggested'] = {}
-            for x in schemas:
-                f['suggested'][x] = schemas[x].is_valid(f)
-        analysis['files'] = files
+    for f in files:
+        f['suggested'] = {}
+        for x in schemas:
+            f['suggested'][x] = schemas[x].is_valid(f)
```

Comment:
This is a weird change, but ok.

Reply:
We only need to provide one container's file suggestions at a time so it was modified to take an array of files rather than several named arrays of containers containing files.